src/main/kotlin/com/playwright/remote/engine/route/UrlMatcher.kt

```diff
@@ -1,4 +1,9 @@
 package com.playwright.remote.engine.route
 
-class UrlMatcher {
+class UrlMatcher(
+    private val rawSource: Any,
+    private val predicate: (String) -> Unit
```

Comment:
Predicate should be of type `(String) -> Boolean`
In your case here it is a consumer of String type.